deepseg.cc

```diff
@@ -371,20 +379,35 @@ void calc_mask(calcinfo_t &info, timinginfo_t &ti) {
 			fprintf(stderr, "Unknown model type\n");
 			break;
 	}
-	ti.maskns=timestamp();
+	info.pti->ai_postns = diffnanosecs(timestamp(), t0);
+	t0 = timestamp();
 
 	// scale up into full-sized mask
-	cv::Mat tmpbuf;
-	cv::resize(info.ofinal,tmpbuf,cv::Size(info.raw.rows/info.ratio,info.raw.rows));
+	{
+		cv::Mat tmpbuf;
+		std::lock_guard<std::mutex> hold(info.lock_mask);
+		cv::resize(info.ofinal,tmpbuf,cv::Size(info.raw_current->rows/info.ratio,info.raw_current->rows));
+		// blur at full size for maximum smoothness
+		cv::blur(tmpbuf,info.mroi,info.blur);
+	}
 
-	// blur at full size for maximum smoothness
-	cv::blur(tmpbuf,info.mroi,info.blur);
+	info.pti->ai_maskns = diffnanosecs(timestamp(), t0);
+	info.pti->ai_totalns = diffnanosecs(timestamp(), t);
 }
 
 bool is_number(const std::string &s) {
   return !s.empty() && std::all_of(s.begin(), s.end(), ::isdigit);
 }
 
+void *calc_mask_thread(void *args) {
+	calcinfo_t *calcinfo = (calcinfo_t *)args;
+	while (!calcinfo->running) usleep(10000);
```

Comment:
Please avoid having code on the same line as the beginning of a block statement.
Feel free to skip braces for single statements though.

Reply:
fixed